### PR TITLE
docs: prepare v0.3.0 release

### PR DESCRIPTION
**Pull request by lvlcn-t**
## Motivation

To adjust our docs to the new check configurations.

## Changes

I've adjusted the `health` and `latency` documentation in the readme and added a link to the readme docs to the helm values for the `checkConfig`.

For additional information look at the commits.

## Tests done

None needed.

Pre-commit passed.

## TODO

- [x] I've assigned this PR to myself
- [x] I've labeled this PR correctly

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -288,17 +291,30 @@ which is named after the DNS name of the `sparrow`. The state file contains the
...
-  the remote sparrow, e.g. `https://sparrow-dns.telekom.de/checks/health`.
-
-Example configuration:
+- `checks`
+  - `health`
+    - `interval` (duration): Interval to perform the health check.
+    - `timeout` (duration): Timeout for the health check.
+    - `retry`
+      - `count` (integer): Number of retries for the health check.
+      - `delay` (duration): Delay between retries for the health check.
```
Comment (puffitos) [inline README.md:300]: Initial delay. We're using the exponential back off, right?

**Comment by y-eight** [comment]
As I mentioned in https://github.com/caas-team/sparrow/pull/76

I would like to correct the docu for the Helm part in the main README as well. I will add it to this PR if you don't mind.

**Review by y-eight — APPROVED** [review]
docu updated

File: README.md [hunk 2]
```diff
@@ -223,20 +233,19 @@ targetManager:
 
 #### Loader
 
-The loader component of the `sparrow` will load the [Runtime](#runtime) configuration dynamically.
+The loader component of the `sparrow` will load the [checks](#checks) configuration dynamically during the runtime.
 
 The loader can be selected by specifying the `loaderType` configuration parameter.
 
-The default loader is an `http` loader that is able to get the runtime configuration from a remote endpoint.
+The default loader is an `http` loader that is able to get the checks configuration from a remote endpoint during the runtime.
```
Comment (lvlcn-t) [inline README.md:240]: is a `http` loader

File: README.md [hunk 3]
```diff
@@ -223,20 +233,19 @@ targetManager:
...
 
 Available loader:
 
 - `http`: The default. Loads configuration from a remote endpoint. Token authentication is available. Additional
-  configuration parameters have the prefix `loaderHttp`.
-- `file` (experimental): Loads configuration once from a local file. Additional configuration parameters have the
-  prefix `loaderFile`. This is just for development purposes.
+  configuration parameters will be set in the `loader.http` section.
+- `file` (experimental): Loads configuration once from a local file. Additional configuration parameters will be set in the `loader.file` section. This is just for development purposes & is currently read just once. The target manager is not supported for this setup!
```
Comment (lvlcn-t) [inline README.md:246]: What do you mean by "is not supported for this setup"? 

If you mean configuring the target manager on runtime, can you rephrase the sentence? It only confuses the user by thinking the target manager is configured on runtime.

File: README.md [hunk 4]
```diff
@@ -223,20 +233,19 @@ targetManager:
...
+The loader component of the `sparrow` will load the [checks](#checks) configuration dynamically during the runtime.
 
 The loader can be selected by specifying the `loaderType` configuration parameter.
 
-The default loader is an `http` loader that is able to get the runtime configuration from a remote endpoint.
+The default loader is an `http` loader that is able to get the checks configuration from a remote endpoint during the runtime.
 
 Available loader:
 
 - `http`: The default. Loads configuration from a remote endpoint. Token authentication is available. Additional
```
Comment (lvlcn-t) [inline README.md:244]:
```md
- `http` (default): Loads ...
```

File: README.md [hunk 5]
```diff
@@ -129,9 +136,12 @@ e.g. `docker run -v /config:/config  ghcr.io/caas-team/sparrow --config /config/
...
 ## Configuration
 
-The configuration is divided into two parts. The startup configuration and the runtime configuration. The startup
-configuration is a technical configuration to configure the `sparrow` instance itself. The runtime configuration will be
-loaded by the `loader` from a remote endpoint. This configuration consists of the checks' configuration.
+The configuration is divided into two parts. The startup configuration and the checks configuration. The startup
+configuration is a technical configuration to configure the `sparrow` instance itself.
+
+The checks configuration will be
+loaded during the runtime by the `loader` from a remote endpoint. The `loader` can be used to get the configuration local as well (type: `file`).
```
Comment (lvlcn-t) [inline README.md:143]: during runtime

File: README.md (hunk 3 above)
Comment (y-eight) [reply README.md]: The reconciling is only triggered when a new config is sent to the config channel https://github.com/caas-team/sparrow/blob/c4a6b545c1b04d871c7e366f1604cc3d36bc1729/pkg/sparrow/run.go#L124 . We need to change that in the future.
Since the `file` loader is currently loading the local config file just once at startup the config never gets reconciled again. The target manager is not able to set the remote targets and will not be passed to the checks. We should change the `file` loader to load the config by an interval as well.

Comment (y-eight) [reply README.md]: I have updated the docu here... hopefully it is more clear now

Comment (lvlcn-t) [reply README.md]: Thank you for clearing this up.

We should indeed add this feature to the file loader but imo it has minor priority.

File: README.md [hunk 6]
```diff
@@ -88,20 +91,27 @@ helm -n sparrow upgrade -i sparrow oci://ghcr.io/caas-team/charts/sparrow --vers
...
+  loader:
+    type: http
+    http:
+      url: https://url-to-checks-config.de/api/config%2Eyaml
 
-runtimeConfig: { }
+checksConfig: {}
 ```
 
+Sensitive data like the token used by the `http` loader (`loader.http.token`) should not be created in the values section. A secret needs to be created manually with a defined environment variable `SPARROW_LOADER_HTTP_TOKEN`. Use the `envFromSecrets` in the `values.yaml` to make the secret accessible by the sparrow container.
```
Comment (puffitos) [inline README.md:111]: To provide the sparrow container with the token (use active and not passive wording, it's always simpler 😉)

File: README.md [hunk 7]
```diff
@@ -126,9 +136,12 @@ e.g. `docker run -v /config:/config  ghcr.io/caas-team/sparrow --config /config/
...
 ## Configuration
 
-The configuration is divided into two parts. The startup configuration and the runtime configuration. The startup
-configuration is a technical configuration to configure the `sparrow` instance itself. The runtime configuration will be
-loaded by the `loader` from a remote endpoint. This configuration consists of the checks' configuration.
+The configuration is divided into two parts. The startup configuration and the checks configuration. The startup
+configuration is a technical configuration to configure the `sparrow` instance itself.
+
+The checks configuration will be
+loaded during runtime by the `loader` from a remote endpoint. The `loader` can be used to get the configuration local as well (type: `file`).
```
Comment (puffitos) [inline README.md:143]: "using a `file` loader, you may load the checks configuration for a local file directly." - let's try to use use direct wording istread of passive voice

File: README.md [hunk 8]
```diff
@@ -220,20 +233,19 @@ targetManager:
 
 #### Loader
 
-The loader component of the `sparrow` will load the [Runtime](#runtime) configuration dynamically.
+The loader component of the `sparrow` will load the [checks](#checks) configuration dynamically during runtime.
 
 The loader can be selected by specifying the `loaderType` configuration parameter.
 
-The default loader is an `http` loader that is able to get the runtime configuration from a remote endpoint.
+The default loader is a `http` loader that is able to get the checks configuration from a remote endpoint during runtime.
```
Comment (puffitos) [inline README.md:240]: An was correct, because http is pronounced eyyyhttp, so it's like it starts with a vowel. Or just use "the default type of the loader is `http`, which is able..."

Comment (puffitos) [reply README.md:240]: Also, "the checks' configuration"

**Review by puffitos — COMMENTED** [review]
Just a couple of grammar corrections and phrasing suggestions.

**Comment by lvlcn-t** [comment]
@puffitos @y-eight I've updated and rephrased some sections of the readme again to improve clarity and readability a little bit to make the content more direct and user-friendly. 

I've made the installation instructions more clear, refined the explanation of the Target Manager, and ensured consistent and concise language throughout.

Can you look over my new commits again to improve possible grammar errors, typos etc?

File: README.md [hunk 9]
```diff
@@ -81,27 +83,33 @@ dedicated [release](https://github.com/caas-team/sparrow/releases) can be found
...
-Sparrow can be installed via Helm Chart. The chart is provided in the GitHub registry:
+Sparrow can be installed via Helm Chart. The chart is available in the GitHub registry:
 
 ```sh
 helm -n sparrow upgrade -i sparrow oci://ghcr.io/caas-team/charts/sparrow --version 1.0.0 --create-namespace
 ```
 
-The default settings are fine for a local running configuration. With the default Helm values, the sparrow loader uses a
-runtime configuration that is provided in a ConfigMap. The ConfigMap can be set by defining the `runtimeConfig` section.
+The default settings are suitable for a local configuration. With the default Helm values, the sparrow loader uses a checks configuration provided in a ConfigMap (the `file` loader is used). Define the `checksConfig` section to set the ConfigMap.
```
Comment (puffitos) [inline README.md:92]: Checks' configurations _or_ a configuration file for the checks

File: README.md [hunk 10]
```diff
@@ -81,27 +83,33 @@ dedicated [release](https://github.com/caas-team/sparrow/releases) can be found
...
 
-To be able to load the configuration during the runtime dynamically, the sparrow loader needs to be set to type `http`.
+To be able to load the checks configuration during the runtime dynamically, the sparrow loader needs to be set to type `http`.
```
Comment (puffitos) [inline README.md:94]: The information provided here is correct and grammatically and syntactically correct.

Still, it's too much info on how to just install via helm. The readme is already too "heavy", we should place this information elsewhere. A good place would be the chart's readme, as a comment over the related entries in the values.yaml file.

File: README.md [hunk 11]
```diff
@@ -126,9 +134,11 @@ e.g. `docker run -v /config:/config  ghcr.io/caas-team/sparrow --config /config/
 
 ## Configuration
 
-The configuration is divided into two parts. The startup configuration and the runtime configuration. The startup
-configuration is a technical configuration to configure the `sparrow` instance itself. The runtime configuration will be
-loaded by the `loader` from a remote endpoint. This configuration consists of the checks' configuration.
+The configuration is divided into two parts. The startup configuration and the checks configuration. The startup
```
Comment (puffitos) [inline README.md:137]: Checks' _or_ configuration for the checks

File: README.md [hunk 12]
```diff
@@ -126,9 +134,11 @@ e.g. `docker run -v /config:/config  ghcr.io/caas-team/sparrow --config /config/
...
+configuration is a technical configuration to configure the `sparrow` instance itself.
+
+The `loader` fetches the checks configuration during runtime from a remote endpoint. For local use, you may directly load the configuration using a `file` loader.
```
Comment (puffitos) [inline README.md:140]: I would first explain what the config for the checks is and then mention how it is loaded. We could also only mention that in the specific section for the checkconfig.

Comment (lvlcn-t) [reply README.md]: I've moved this [here](https://github.com/caas-team/sparrow/pull/76/files/5d88dff438baf80531224eef455c366e2dba59c8..e4aed56a4bfb9843c1b0976b873c42f99ffb1ecc#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R241)

File: README.md (hunk 10 above)
Comment (y-eight) [reply README.md]: We could place several config options in the values.yaml with docu blocks

File: README.md [hunk 13]
```diff
@@ -288,17 +289,30 @@ which is named after the DNS name of the `sparrow`. The state file contains the
...
-  another `sparrow` instance. Automatically used when target manager is activated otherwise use the health endpoint of
-  the remote sparrow, e.g. `https://sparrow-dns.telekom.de/checks/health`.
+- `checks`
+  - `health`
+    - `interval` (duration): Interval to perform the health check.
+    - `timeout` (duration): Timeout for the health check.
+    - `retry`
+      - `count` (integer): Number of retries for the health check.
+      - `delay` (duration): Initial delay between retries for the health check.
+    - `targets` (list of strings): List of targets to send health probe. Needs to be a valid url. Can be
```
[the remark this answers is not in the document]
Comment (puffitos) [reply README.md:293]: This is still open @lvlcn-t - please change to URL.

File: README.md (hunk 10 above)
Comment (lvlcn-t) [reply README.md]: I've put this now into the [startup section](https://github.com/caas-team/sparrow/pull/76/commits/4582c2235597918a7e73d3c2452ba4426ab3773b).

**Review by puffitos — APPROVED** [review]
Fine by me.
